### PR TITLE
refactor(rome_js_parser): refactor regex flag

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
## Summary
1. Refactor `lexer.rs` lexing regular expression, using a `bitflags` to represent regex flag instead of 7 boolean variables.
Reducing memory usage from 7 bytes to 1 byte.

## Test Plan
1. All CI should pass.

### CODE REVIEW

File: crates/rome_js_parser/src/lexer/mod.rs [hunk 1]
```diff
@@ -1555,53 +1565,59 @@ impl<'src> Lexer<'src> {
...
-                                    s = true;
                                 }
                                 b'u' => {
-                                    if u {
+                                    if flag.contains(RegexFlag::U) {
                                         self.diagnostics.push(self.flag_err('u'));
+                                    } else {
+                                        flag |= RegexFlag::U;
                                     }
-                                    u = true;
```
Comment (IWANABETHATGUY) [inline crates/rome_js_parser/src/lexer/mod.rs:1592]: This statement will be always executed before, using an else branch to avoid that.

**Comment by IWANABETHATGUY** [comment]
!bench_parser

File: crates/rome_js_parser/src/lexer/mod.rs [hunk 2]
```diff
@@ -1530,10 +1530,20 @@ impl<'src> Lexer<'src> {
             "this flag was already used",
         )
     }
-
     #[inline]
     #[allow(clippy::many_single_char_names)]
     fn read_regex(&mut self) -> JsSyntaxKind {
+        bitflags! {
+            struct RegexFlag: u32 {
```
Comment (MichaReiser) [inline crates/rome_js_parser/src/lexer/mod.rs:1537]: You can reduce this to be a u8 (you only have 7 flags)

File: crates/rome_js_parser/src/lexer/mod.rs (hunk 1 above)
Comment (MichaReiser) [reply crates/rome_js_parser/src/lexer/mod.rs:1592]: Adding the same flag has no effect, which is why I would probably prefer to keep the assignment outside of the else branch.

**Comment by IWANABETHATGUY** [comment]
!bench_parser

**Review by MichaReiser — APPROVED** [review]

**Pull request merged**